### DISCUSSION TITLE
Regular expressions are case insensitive by default

### DISCUSSION THREAD

**Discussion by maruel** [community/community#11130] [Other Feature Feedback, Questions, & Ideas] [answered]
Defaulting to case-insensitive search for regular expression is surprising. I would expect regexp statements to be case sensitive unless explicitly opted-in.

As an example, `/Reed-Solmon/` hits `reed-solomon`.

The documentation https://cs.github.com/about/syntax#regex is sparse to say the least. Is it using re2? What notation does it support? Thanks.

**Comment by colinwm** [comment, marked as answer]
Hey @maruel, thanks for the feedback. Right now regular expressions are case insensitive because our search index is case insensitive, which means that case sensitive search is somewhat inefficient. We're working on making case sensitivity possible, though.

As for the regex engine, we're using this: https://docs.rs/regex/latest/regex/, so whatever that engine can parse we should be able to search.

**Reply by maruel** [reply]
Thanks. Can you put a link to https://docs.rs/regex/latest/regex/#syntax at https://cs.github.com/about/syntax#regex?

Also there's something weird with the page that https://cs.github.com/about/syntax#regex doesn't navigate to the "Regular expressions" header.

**Comment by 3052** [comment]
this seems to work:

```
/(?-i)PDS/
```

https://github.com/orgs/community/discussions/9759
